tools/fork-network/src/cli.rs

```diff
@@ -85,9 +85,43 @@ enum SubCommand {
 
 #[derive(clap::Parser)]
 struct InitCmd {
-    /// If given, the shard layout in this file will be used to generate the forked genesis state
-    #[arg(long)]
-    pub shard_layout_file: Option<PathBuf>,
+    /// Shard layout override option. Available options:
+    /// - no_override: Use the shard layout from the source chain
+    /// - file:<path>: Use the shard layout defined in the given file
+    /// - pv:<version>: Use the shard layout from the given protocol version
+    #[clap(long, value_parser = parse_shard_override, default_value = "no_override")]
+    pub shard_layout_override: ShardLayoutOverride,
+}
+
+#[derive(Clone, Debug)]
+enum ShardLayoutOverride {
+    NoOverride,
+    UseShardLayoutFromFile(PathBuf),
+    UseShardLayoutFromProtocolVersion(ProtocolVersion),
+}
+
+// Custom parser for the enum because some variants carry data.
+fn parse_shard_override(s: &str) -> Result<ShardLayoutOverride, String> {
```

Comment:
nit: maybe you can use `groups` to parse arguments in a more natural way [ref](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=e881a31789bcaf0302949ecf08eec6cd)